### ISSUE TITLE
Can't change search strategy for existing index 

### ISSUE THREAD

**Issue by evgeniagusakova** [strapdata/elassandra#306]
Hi !
Elassandra version: 6.2.3.17
OS version: Ubuntu 18.04.2 LTS
```
$ java -version
openjdk version "1.8.0_222"
OpenJDK Runtime Environment (build 1.8.0_222-8u222-b10-1ubuntu1~18.04.1-b10)
OpenJDK 64-Bit Server VM (build 25.222-b10, mixed mode)
```

I'm trying to change search strategy for existing index in the following way:
```
curl -XPUT -H "Content-Type: application/json" "http://hostname:9200/index_name/_settings" -d '{ "settings" : { "index.search_strategy_class":"RandomSearchStrategy" } }'
```

And getting the following error:
```
{
  "error": {
    "root_cause": [
      {
        "type": "illegal_state_exception",
        "reason": "rejecting cluster state version [205] uuid [lDkCWhnVRMWTts3x7uMrmA] received from [f9ed38c8-b7e0-4944-a0fe-4af6b5c6eb84]"
      }
    ],
    "type": "illegal_state_exception",
    "reason": "rejecting cluster state version [205] uuid [lDkCWhnVRMWTts3x7uMrmA] received from [f9ed38c8-b7e0-4944-a0fe-4af6b5c6eb84]"
  },
  "status": 500
}
```
In logs I see the following

```
2019-09-16 17:42:14,482 WARN  [elasticsearch[hostname][masterService#updateTask][T#1]] CassandraDiscovery.java:1100 onNewClusterStateFailed failed while applying cluster state locally source=update-settings
java.lang.IllegalStateException: rejecting cluster state version [205] uuid [lDkCWhnVRMWTts3x7uMrmA] received from [f9ed38c8-b7e0-4944-a0fe-4af6b5c6eb84]
	at org.elassandra.discovery.CassandraDiscovery.processNextCommittedClusterState(CassandraDiscovery.java:1196)
	at org.elassandra.discovery.CassandraDiscovery.publishLocalUpdate(CassandraDiscovery.java:1110)
	at org.elassandra.discovery.CassandraDiscovery.publish(CassandraDiscovery.java:883)
	at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:251)
	at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:152)
	at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150)
	at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188)
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:573)
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:244)
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:207)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

I successfully changed search strategy with the same request on Elassandra 5.5.0.24. According to the documentation https://elassandra.readthedocs.io/en/latest/operations.html?highlight=randomsearchstrategy#optimizing-search-requests it should work.

Could you please help me with changing  of the search strategy ?

**Comment by vroyer** [comment]
Hi Evgenia,

You probably have another error in your logs (or a nested exception) explaining why the cluster state is rejected ?

**Comment by evgeniagusakova** [comment]
@vroyer Thanks for the reply!


I've started tail command before curl execution (curl command is the same as I posted above) and I see in logs:
```
sudo tail -f /var/log/cassandra/system.log

2019-09-16 18:28:08,064 WARN  [elasticsearch[hostname][masterService#updateTask][T#1]] BytesRestResponse.java:133 build path: /index_name/_settings, params: {index=stories_qa}
java.lang.IllegalStateException: rejecting cluster state version [206] uuid [BbRJFOdvQX68zsqIW8jefg] received from [f9ed38c8-b7e0-4944-a0fe-4af6b5c6eb84]
  at org.elassandra.discovery.CassandraDiscovery.processNextCommittedClusterState(CassandraDiscovery.java:1196)
  at org.elassandra.discovery.CassandraDiscovery.publishLocalUpdate(CassandraDiscovery.java:1110)
  at org.elassandra.discovery.CassandraDiscovery.publish(CassandraDiscovery.java:883)
  at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:251)
  at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:152)
  at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150)
  at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188)
  at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:573)
  at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:244)
  at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:207)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
  at java.lang.Thread.run(Thread.java:748)
2019-09-16 18:28:08,065 WARN  [elasticsearch[hostname][masterService#updateTask][T#1]] CassandraDiscovery.java:1100 onNewClusterStateFailed failed while applying cluster state locally source=update-settings
java.lang.IllegalStateException: rejecting cluster state version [206] uuid [BbRJFOdvQX68zsqIW8jefg] received from [f9ed38c8-b7e0-4944-a0fe-4af6b5c6eb84]
  at org.elassandra.discovery.CassandraDiscovery.processNextCommittedClusterState(CassandraDiscovery.java:1196)
  at org.elassandra.discovery.CassandraDiscovery.publishLocalUpdate(CassandraDiscovery.java:1110)
  at org.elassandra.discovery.CassandraDiscovery.publish(CassandraDiscovery.java:883)
  at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:251)
  at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:152)
  at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150)
  at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188)
  at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:573)
  at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:244)
  at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:207)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
  at java.lang.Thread.run(Thread.java:748)
```

**Comment by vroyer** [comment]
I reproduced your issue, it's a bug when updating the index settings.
It's fixed for the next release 6.2.3.19.

**Comment by evgeniagusakova** [comment]
@vroyer Thank you for the fast reply!

I'll check and update this ticket as soon as 6.2.3.19 will be released.

**Comment by vroyer** [comment]
This is now fixed in release 6.2.3.19.
Thanks for reporting the issue.
